File: src/main/java/com/trade/securities/application/service/EtnTradingInfoService.java
package com.trade.securities.application.service;

import com.trade.securities.application.port.in.GetEtnTradingInfoUseCase;
import com.trade.securities.application.port.out.LoadEtnTradingInfoPort;
import com.trade.securities.domain.EtnTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class EtnTradingInfoService implements GetEtnTradingInfoUseCase {

    private final LoadEtnTradingInfoPort loadEtnTradingInfoPort;

    @Override
    public List<EtnTradingInfo> getEtnTradingInfo(String basDd) {
        return loadEtnTradingInfoPort.loadEtnTradingInfo(basDd);
    }
}

File: src/main/kotlin/com/trade/securities/application/service/EtnTradingInfoService.kt
package com.trade.securities.application.service

import com.trade.securities.application.port.`in`.GetEtnTradingInfoUseCase
import com.trade.securities.application.port.out.LoadEtnTradingInfoPort
import com.trade.securities.domain.EtnTradingInfo
import org.springframework.stereotype.Service

@Service
class EtnTradingInfoService(
    private val loadEtnTradingInfoPort: LoadEtnTradingInfoPort
) : GetEtnTradingInfoUseCase {

    override fun getEtnTradingInfo(basDd: String): List<EtnTradingInfo> {
        return loadEtnTradingInfoPort.loadEtnTradingInfo(basDd)
    }
}
